src/CommandingOfficers/Commander.java

```diff
@@ -384,4 +394,66 @@ public GameAction getNextAIAction(GameMap gameMap)
     }
     return null;
   }
+
+  /**
+   * Private method, same signature as in Serializable interface
+   *
+   * @param stream
+   * @throws IOException
+   */
+  private void writeObject(ObjectOutputStream stream) throws IOException
```

Comment:
It appears that this function (and readObject below) are never used. After deleting them, I was still able to save/reload a game.

Reply:
Try saving and reloading a game with an AI in it.

I suspect it might prevent you from buying new troops, as well.

Reply:
Hmm, true enough. Better question. Why declare `unitProductionByTerrain` as `transient` and then serialize a representation of its contents manually?

Reply:
Hm. Is it because the `TerrainType` objects used as keys are static, so Java ignores them? That's... inconvenient.

Reply:
I'm pretty sure they're not ignored, but deserialized data is a *copy*, by definition. That makes all of our identity comparisons moot.

I actually ran into the same problem with strings. My hack to prevent ability energy from accumulating during powers was to check if your active ability string was equal to the empty string. Worked marvelously. Until save/reload.
Turns out it was a copy of the empty string, so "" != "" and nobody could *ever* charge powers.

Reply:
To answer the static question, it doesn't matter.
The map that I'm storing is *not* static, so it's irrelevant that the value I'm referencing happens to be defined statically.
I can only assume that the reference coming *out* of serialization would be shared by all COs that were deserialized, but that doesn't help because it still won't match the statically defined copies (this holds even if you don't exit the game, as deserialized objects are copies.).